interpreter/VM.c

```diff
@@ -1190,7 +1190,11 @@ void vm_close_all(struct VM *const vm) {
 static void vm_ECHO(struct VM *const vm) {
 	vm_stringify_top(vm);
 	struct YASL_String *v = vm_popstr(vm);
-	vm_print_out(vm, "%.*s\n", (int)YASL_String_len(v), v->start + v->str);
+	size_t strlen = (int)YASL_String_len(v);
+	char *dest = (char *) malloc(strlen);
+	size_t copied = io_str_strip_char(dest, v->str, strlen, 0);
```

Comment:
This should be `v->str + v->start`, `v->str` points to the beginning of the underlying buffer, not the start of the string.

You can see that if you do a slice for example:
```
const x = '\0abc\0def'

echo len x

echo x
echo x[1:]

echo len x

```

Reply:
k, changing it iab

Reply:
Done ;)